sotodlib/hwp/g3thwp.py

```diff
@@ -322,145 +424,73 @@ def analyze(self, data, ratio=0.25, mod2pi=True, fast=True):
 
         if not any(data):
             logger.info("no HWP field data")
-        ## Analysis parameters ##
-        # Fast block
-        counter = np.array([])
-        counter_idx = np.array([])
-        quad_time = np.array([])
-        quad = np.array([])
-        if 'counter' in data.keys():
-            counter = data['counter'][1]
-            counter_idx = data['counter_index'][1]
-            quad_time = data['quad'][0]
-            quad = self._quad_form(data['quad'][1])
-
-        irig_time = np.array([])
-        rising_edge = np.array([])
-        if 'irig_time' in data.keys():
-            irig_time = data['irig_time'][1]
-            rising_edge = data['rising_edge_count'][1]
-            if self._irig_type == 1:
-                irig_time = data['irig_synch_pulse_clock_time'][1]
-                rising_edge = data['irig_synch_pulse_clock_counts'][1]
-            logger.info('IRIG timing quality check.')
-            irig_time, rising_edge = self._irig_quality_check(
-                irig_time, rising_edge)
-            if len(irig_time) == 0:
-                logger.warning('All IRIG time is not correct in 1st encoder')
-
-        irig_time_2 = np.array([])
-        rising_edge_2 = np.array([])
+
+        d = self._data_formatting(data)
         if 'irig_time_2' in data.keys() and 'counter_2' in data.keys():
-            logger.info('checking 2nd encoder.')
-            irig_time_2 = data['irig_time_2'][1]
-            rising_edge_2 = data['rising_edge_count_2'][1]
-            if self._irig_type == 1:
-                irig_time_2 = data['irig_synch_pulse_clock_time_2'][1]
-                rising_edge_2 = data['irig_synch_pulse_clock_counts_2'][1]
-            irig_time_2, rising_edge_2 = self._irig_quality_check(
-                irig_time_2, rising_edge_2)
-            if len(irig_time_2) == 0:
-                logger.info('All IRIG time is not correct in 2nd encoder')
-
-        if len(irig_time) == 0 and len(irig_time_2) == 0:
+            d2 = self._data_formatting(data, suffix='_2')
+            if len(d['irig_time']) < len(d2['irig_time']):
+                logger.info('Use 2nd encoder.')
+                d = d2
+        if len(d['irig_time']) == 0:
             logger.warning('There is no correct IRIG timing. Stop analyze.')
             return {}
-        if len(irig_time) < len(irig_time_2):
-            logger.info('Use 2nd encoder IRIG timing.')
-            irig_time = irig_time_2
-            rising_edge = rising_edge_2
-            counter = data['counter_2']
-            counter_index = data['counter_index_2']
-            quad_time = data['quad_2'][0]
-            quad = self._quad_form(data['quad_2'][1])
```

Comment:
comment about small bug I found.

Previously there was a small bug of using 2nd encoder as follows
`data['counter_2'] =>data['counter_2'][1]` etc